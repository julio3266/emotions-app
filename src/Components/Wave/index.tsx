import React from 'react';

import {Svg, Path} from 'react-native-svg';
import Animated from 'react-native-reanimated';
import {
  GestureEvent,
  PanGestureHandler,
  PanGestureHandlerEventPayload,
} from 'react-native-gesture-handler';
import {assignTestId} from '@utils/qualityAssurance';

export interface IWaveAnimation {
  gestureWaveAnimated: (
    event: GestureEvent<PanGestureHandlerEventPayload>,
  ) => void;
  waveAnimationStyle: any;
}

export interface IWave {
  testID?: string;
  width: number;
  height: number;
  color: string;
  animationProps: IWaveAnimation;
}

export const Wave: React.FC<IWave> = ({
  testID = 'wave',
  width,
  height,
  color,
  animationProps,
}) => {
  const staticWave =
    'M98 12.072C116.965 26.7798 139 34.072 150.5 33.072V44.572H0V31.572C9.5 26.072 11.4375 25.572 18 18.072C35.9376 -2.428 73.5 -6.92799 98 12.072Z';

  return (
    <PanGestureHandler
      {...assignTestId('View', testID)}
      onGestureEvent={animationProps.gestureWaveAnimated}>
      <Animated.View
        {...assignTestId('View', `${testID}_svgContainer`)}
        style={animationProps.waveAnimationStyle}>
        <Svg
          {...assignTestId('View', `${testID}_svg`)}
          width={width}
          height={height}
          viewBox="0 0 151 45"
          fill="none">
          <Path d={staticWave} fill={color} />
        </Svg>
      </Animated.View>
    </PanGestureHandler>
  );
};
