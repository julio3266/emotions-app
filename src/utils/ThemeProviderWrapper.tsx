import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
