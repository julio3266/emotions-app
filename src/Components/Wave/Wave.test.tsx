import React from 'react';
import {ThemeProviderWrapper as wrapper} from '@utils/ThemeProviderWrapper';
import {mockTestID} from '@utils/qualityAssurance';
import {render} from '@testing-library/react-native';
import {Wave, IWave} from '@Components/Wave';
import Theme from '@theme/styles';

describe('Components/Wave', () => {
  let props: IWave = {
    testID: 'wave',
    animationProps: {
      gestureWaveAnimated: jest.fn(),
      waveAnimationStyle: [],
    },
    color: Theme.colors.blueViolet,
    height: 40,
    width: 100,
  };

  const {findByTestId, toJSON} = render(<Wave {...props} />, {
    wrapper,
  });
  const sut_container = mockTestID('View', props.testID!).testID;
  const el_svgContainer = mockTestID(
    'View',
    `${props.testID!}_svgContainer`,
  ).testID;

  const el_svg = mockTestID('View', `${props.testID!}_svg`).testID;

  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_svgContainer}`, () => {
    const sut = findByTestId(el_svgContainer);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_svg}`, () => {
    const sut = findByTestId(el_svg);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
