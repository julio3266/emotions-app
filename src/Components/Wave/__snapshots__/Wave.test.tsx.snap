// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Wave should render a snapshot 1`] = `null`;
